File: classified/packages/game/cypress/support/e2e.ts
// ***********************************************************
// This example support/e2e.ts is processed and
// loaded automatically before your test files.
//
// This is a great place to put global configuration and
// behavior that modifies Cypress.
//
// You can change the location of this file or turn off
// automatically serving support files with the
// 'supportFile' configuration option.
//
// You can read more here:
// https://on.cypress.io/configuration
// ***********************************************************

// Import commands.js using ES2015 syntax:
import './commands';
import './test-commands';

// Global configuration
Cypress.config('defaultCommandTimeout', 15000);
Cypress.config('requestTimeout', 20000);
Cypress.config('responseTimeout', 20000);
Cypress.config('pageLoadTimeout', 30000);

// Global error handling
Cypress.on('uncaught:exception', (err, _runnable) => {
  // Don't fail tests on uncaught exceptions unless they're test-related
  if (err.message.includes('ResizeObserver loop limit exceeded')) {
    return false; // Ignore ResizeObserver errors
  }

  if (err.message.includes('Non-Error promise rejection captured')) {
    return false; // Ignore promise rejection warnings
  }

  if (err.message.includes('Script error')) {
    return false; // Ignore script errors from external sources
  }

  // Log other errors but don't fail the test
  console.error('Uncaught exception:', err);
  return false;
});

// Global setup for API authentication
beforeEach(() => {
  // Set up X-API-KEY header for all requests if auth token is available
  const authToken = Cypress.env('ELIZA_SERVER_AUTH_TOKEN');
  if (authToken) {
    cy.intercept('**', (req) => {
      // Only add header to API requests to our backend
      if (req.url.includes('/api/') && req.url.includes('localhost:7777')) {
        req.headers['X-API-KEY'] = authToken;
      }
    });
  }
});

// Global before hook
beforeEach(() => {
  // Set up consistent test environment
  cy.viewport(1280, 720);

  // Clear any previous state
  cy.clearLocalStorage();
  cy.clearCookies();

  // Set up test environment
  cy.window().then((win) => {
    win.localStorage.setItem('testingMode', 'true');
    win.localStorage.setItem('cypressTest', 'true');
  });

  // Intercept and log all network requests
  cy.intercept('**/*', (req) => {
    console.log(`Network request: ${req.method} ${req.url}`);
    req.continue();
  });
});

// Global after hook
afterEach(() => {
  // Capture final state for debugging
  cy.window().then((_win) => {
    // Just log that the test completed
    console.log('Test completed at:', new Date().toISOString());
  });

  // Take final screenshot
  cy.screenshot('test-completed');

  // Clean up any remaining test data
  // cy.cleanupTestData(); // Disabled for now - UI elements not present
});

// Add custom assertion
chai.use((chai, utils) => {
  utils.addMethod(chai.Assertion.prototype, 'containOneOf', function (list) {
    const obj = utils.flag(this, 'object');
    const found = list.some((item) => obj.includes(item));

    this.assert(
      found,
      `expected "${obj}" to contain one of [${list.join(', ')}]`,
      `expected "${obj}" not to contain any of [${list.join(', ')}]`
    );
  });
});

// Extend Cypress namespace for TypeScript
declare global {
  namespace Cypress {
    interface Chainable {
      waitForAgentReady(): Chainable<void>;
      sendMessageAndWait(message: string): Chainable<void>;
      toggleCapability(capability: string, expectedState: boolean): Chainable<void>;
      verifyErrorHandling(errorType: string): Chainable<void>;
      resetAppState(): Chainable<void>;
      enableTestingMode(): Chainable<void>;
      simulateNetworkCondition(condition: 'offline' | 'slow' | 'unstable'): Chainable<void>;
      checkA11y(): Chainable<void>;
      measurePerformance(testName: string): Chainable<void>;
      testResponsive(): Chainable<void>;
      testKeyboardNavigation(): Chainable<void>;
      verifyTextOnScreen(expectedText: string): Chainable<void>;
      cleanupTestData(): Chainable<void>;
      tab(options?: { shift: boolean }): Chainable<Element>;
    }

    interface Assertion {
      containOneOf(list: string[]): Chainable<Element>;
    }
  }
}

File: cpp_generated/classified/packages/game/cypress/support/e2e.cpp
#include "/home/runner/work/elizaos-cpp/elizaos-cpp/classified/packages/game/cypress/support/e2e.h"

namespace global {
    namespace Cypress {
    }
}

void Main(void)
{
    Cypress->config(std::string("defaultCommandTimeout"), 15000);
    Cypress->config(std::string("requestTimeout"), 20000);
    Cypress->config(std::string("responseTimeout"), 20000);
    Cypress->config(std::string("pageLoadTimeout"), 30000);
    Cypress->on(std::string("uncaught:exception"), [=](auto err, auto _runnable) mutable
    {
        if (err["message"]["includes"](std::string("ResizeObserver loop limit exceeded"))) {
            return false;
        }
        if (err["message"]["includes"](std::string("Non-Error promise rejection captured"))) {
            return false;
        }
        if (err["message"]["includes"](std::string("Script error"))) {
            return false;
        }
        console->error(std::string("Uncaught exception:"), err);
        return false;
    }
    );
    beforeEach([=]() mutable
    {
        shared authToken = Cypress->env(std::string("ELIZA_SERVER_AUTH_TOKEN"));
        if (authToken) {
            cy->intercept(std::string("**"), [=](auto req) mutable
            {
                if (AND((req["url"]["includes"](std::string("/api/"))), (req["url"]["includes"](std::string("localhost:7777"))))) {
                    req["headers"][std::string("X-API-KEY")] = authToken;
                }
            }
            );
        }
    }
    );
    beforeEach([=]() mutable
    {
        cy->viewport(1280, 720);
        cy->clearLocalStorage();
        cy->clearCookies();
        cy->window()->then([=](auto win) mutable
        {
            win["localStorage"]["setItem"](std::string("testingMode"), std::string("true"));
            win["localStorage"]["setItem"](std::string("cypressTest"), std::string("true"));
        }
        );
        cy->intercept(std::string("**/*"), [=](auto req) mutable
        {
            console->log(std::string("Network request: ") + req["method"] + std::string(" ") + req["url"] + string_empty);
            req["_continue"]();
        }
        );
    }
    );
    afterEach([=]() mutable
    {
        cy->window()->then([=](auto _win) mutable
        {
            console->log(std::string("Test completed at:"), ((std::make_shared<Date>()))->toISOString());
        }
        );
        cy->screenshot(std::string("test-completed"));
    }
    );
    chai->use([=](auto chai, auto utils) mutable
    {
        utils->addMethod(chai->Assertion->prototype, std::string("containOneOf"), [=](any list) mutable
        {
            shared obj = utils->flag(shared_from_this(), std::string("object"));
            auto found = list["some"]([=](auto item) mutable
            {
                return obj["includes"](item);
            }
            );
            this["assert"](found, std::string("expected "") + obj + std::string("" to contain one of [") + list["join"](std::string(", ")) + std::string("]"), std::string("expected "") + obj + std::string("" not to contain any of [") + list["join"](std::string(", ")) + std::string("]"));
        }
        );
    }
    );
}

MAIN
